src/CSI.IBTA.UserService/Services/UsersService.cs

```diff
@@ -0,0 +1,301 @@
+﻿using CSI.IBTA.DataLayer.Interfaces;
+using CSI.IBTA.Shared.DTOs;
+using CSI.IBTA.Shared.DTOs.Errors;
+using CSI.IBTA.Shared.Entities;
+using CSI.IBTA.Shared.Utils;
+using CSI.IBTA.UserService.Interfaces;
+using System.Net;
+using System.Security.Claims;
+
+namespace CSI.IBTA.UserService.Services
+{
+    internal class UsersService : IUsersService
+    {
+        private readonly IUnitOfWork _unitOfWork;
+
+        public UsersService( 
+            IUnitOfWork unitOfWork)
+        {
+            _unitOfWork = unitOfWork;
+        }
+
+        public async Task<GenericResponse<UserDto[]>> GetAllUsers(HttpContext httpContext)
+        {
+            await _unitOfWork.Accounts.All();
+            await _unitOfWork.Employers.All();
+            var employerUsers = (await _unitOfWork.EmployerUsers.All()).ToArray();
+            var userList = (await _unitOfWork.Users.All())
+                .Select(user => new UserDto(user.Id, user.Account.Username, 
+                user.Firstname, user.Lastname, user.Account.Id, 
+                UserHasEmployer(employerUsers, user)))
+                .ToArray();
+
+            //.Select(user => new UserDto(user.Id, user.Account.Username, 
+            //user.Firstname, user.Lastname, user.Account.Id, employerId));
+            
+            return new GenericResponse<UserDto[]>(false, null, userList);
+        }
+
+        public async Task<GenericResponse<UserDto>> GetUser(int accountId, HttpContext httpContext)
+        {
+            await _unitOfWork.Accounts.All();
```

Comment:
The same as above - relations should be handled by EF mappings, not manually. 
Also, in order to get 1 user, your code queries all Users + all employers + all relations between then. 
If we gonna have 10k users more that would be very slow and expensive operation.